pkgs/build-support/trivial-builders.nix

```diff
@@ -814,4 +814,33 @@ rec {
       ${command} |& grep -Fw ${version}
       touch $out
     '';
+
+  testGraphical =
+    { package
+    , command ? "${package.meta.mainProgram or package.pname or package.name}"
+    , expectedText ? "File" # Very very common in menu bars.
+    }: import ../../nixos/tests/make-test-python.nix ({ pkgs, ... }: {
```

Comment:
You could use `pkgs.nixosTest`, which is a little bit faster as it reuses the existing nixpkgs invocation.